lib/sortjoinstream.js

```diff
@@ -106,9 +106,20 @@ SortJoinStream.prototype._nextTriple = function nextTriple(skip) {
 
 SortJoinStream.prototype._start = function() {
   var that = this;
-
   this._readStream = this.db.getStream(this._queryMask, { index: this.index });
 
+  if(!this._readStream.read){
+    this._readStream._cache = [];
+
+    this._readStream.read = function() {
+      return that._readStream._cache.splice(0, 1)[0];
+    };
+
+    this._readStream.on('data', function(data) {
+      that._readStream._cache.push(data);
+    });
+  }
```

Comment:
in multilevel, rather than in LevelGraph.
Or maybe it's a different fix.
If getStream does not return a stream (as it is missing read()), we can hardly expect consistent behavior.